batch/batch/driver/scheduler.py

```diff
@@ -206,19 +224,11 @@ def allocate_cores(user, mark):
                     instance.adjust_free_cores_in_memory(-record['cores_mcpu'])
                     should_wait = False
                     scheduled_cores_mcpu += record['cores_mcpu']
-                    to_schedule.append((record, instance))
-
-        results = await bounded_gather(*[functools.partial(schedule_job, self.app, record, instance)
-                                         for record, instance in to_schedule],
-                                       parallelism=100,
-                                       return_exceptions=True)
-
-        for ((record, instance), result) in zip(to_schedule, results):
-            batch_id = record['batch_id']
-            job_id = record['job_id']
-            id = (batch_id, job_id)
 
-            if isinstance(result, Exception):
-                log.info(f'error while scheduling job {id} on {instance}, {result!r}')
+                    async_work.append(
+                        LogAndIgnoreErrors(
+                            f'scheduling job {id} on {instance}',
+                            functools.partial(schedule_job, self.app, record, instance)))
 
+        await bounded_gather(*[x.async_action for x in async_work], parallelism=100)
```

Comment:
I don't quite understand how the async_action works. I know it's calling the wrapped function with error handling. Where does `__call__` get used?

Reply:
This was broken. The new version is simpler and works correctly. Exceptions are caught by the jobs themselves instead of post-hoc. This means if 1 job fails and 9 are still waiting, we can run the log statement while we wait for the other 9 instead of waiting to do that at the end.